mod/pubsub.php

```diff
@@ -4,164 +4,134 @@
-			$sql_extra = sprintf(" AND `hub-verify` = '%s' ", dbesc($hub_verify));
-			$log_info = 'hub-verify: ' . $hub_verify;
-		} else {
-			$sql_extra = sprintf(" AND `id` = %d ", intval($contact_id));
-			$log_info = 'contact-id: ' . $contact_id;
+			$condition['hub-verify'] = $hub_verify;
 		}
 
-		$r = q("SELECT * FROM `contact` WHERE `uid` = %d
-			AND NOT `blocked` AND NOT `pending` $sql_extra LIMIT 1",
-			intval($owner['uid'])
-		);
-		if (!DBM::is_result($r)) {
-			logger('pubsub: contact not found. ' . $log_info);
+		$contact = dba::selectFirst('contact', ['id', 'poll'], $condition);
+		if (!DBM::is_result($contact)) {
+			logger('Contact ' . $contact_id . ' not found.');
 			hub_return(false, '');
 		}
 
-		if (!empty($hub_topic)) {
-			if (!link_compare($hub_topic,$r[0]['poll'])) {
-				logger('pubsub: hub topic ' . $hub_topic . ' != ' . $r[0]['poll']);
-				// should abort but let's humour them.
-			}
+		if (!empty($hub_topic) && !link_compare($hub_topic, $contact['poll'])) {
+			logger('Hub topic ' . $hub_topic . ' != ' . $contact['poll']);
+			hub_return(false, '');
 		}
 
-		$contact = $r[0];
-
 		// We must initiate an unsubscribe request with a verify_token.
 		// Don't allow outsiders to unsubscribe us.
 
-		if ($hub_mode === 'unsubscribe') {
-			if (!strlen($hub_verify)) {
-				logger('pubsub: bogus unsubscribe');
-				hub_return(false, '');
-			}
+		if (($hub_mode === 'unsubscribe') && empty($hub_verify)) {
+			logger('Bogus unsubscribe');
+			hub_return(false, '');
 		}
 
 		if (!empty($hub_mode)) {
 			dba::update('contact', ['subhub' => $subscribe], ['id' => $contact['id']]);
-			logger('pubsub: ' . $hub_mode . ' success');
+			logger($hub_mode . ' success for contact ' . $contact_id . '.');
 		}
  		hub_return(true, $hub_challenge);
 	}
 }
 
-require_once('include/security.php');
-
-function pubsub_post(App $a) {
-
+function pubsub_post(App $a)
+{
 	$xml = file_get_contents('php://input');
 
-	logger('pubsub: feed arrived from ' . $_SERVER['REMOTE_ADDR'] . ' for ' .  $a->cmd );
-	logger('pubsub: user-agent: ' . $_SERVER['HTTP_USER_AGENT'] );
-	logger('pubsub: data: ' . $xml, LOGGER_DATA);
+	logger('Feed arrived from ' . $_SERVER['REMOTE_ADDR'] . ' for ' .  $a->cmd . ' with user-agent: ' . $_SERVER['HTTP_USER_AGENT']);
+	logger('Data: ' . $xml, LOGGER_DATA);
 
 	$nick       = (($a->argc > 1) ? notags(trim($a->argv[1])) : '');
 	$contact_id = (($a->argc > 2) ? intval($a->argv[2])       : 0 );
 
-	$r = q("SELECT * FROM `user` WHERE `nickname` = '%s' AND `account_expired` = 0 AND `account_removed` = 0 LIMIT 1",
-		dbesc($nick)
-	);
-	if (!DBM::is_result($r)) {
+	$importer = dba::selectFirst('user', [], ['nickname' => $nick, 'account_expired' => false, 'account_removed' => false]);
+	if (!DBM::is_result($importer)) {
 		hub_post_return();
 	}
 
-	$importer = $r[0];
+	$condition = ['id' => $contact_id, 'uid' => $importer['uid'], 'subhub' => true, 'blocked' => false];
+	$contact = dba::selectFirst('contact', [], $condition);
```

Comment:
Why drop the `network` and `rel` conditions?

Reply:
The check is now done some lines below. We have to check the fetched contact. Having this in the query would lead to some not found contact - which would trigger the second try with fetching the contact via the feed content.